syntax = "proto3";

package org.couchers.api.user_blocks;

import "google/protobuf/empty.proto";

service UserBlocks {

  rpc BlockUser(BlockUserReq) returns (google.protobuf.Empty);

  //Returns all users blocked by a given account
  rpc GetBlockedUsers(google.protobuf.Empty) returns (GetBlockedUsersRes);

  rpc UnblockUser(UnblockUserReq) returns(google.protobuf.Empty);

}

message BlockUserReq {
  int64 user_id = 1;
}

message UnblockUserReq {
  int64 user_id = 1;
}

message GetBlockedUsersRes {
  repeated int64 blocked_user_ids = 1;
}